packages/fa-icon-chooser/src/components/fa-icon/fa-icon.tsx

```diff
@@ -62,103 +62,129 @@ export class FaIcon {
+    const missingProps = [];
+    if (!this.svgApi) missingProps.push('svgApi');
+    if (!this.stylePrefix) missingProps.push('stylePrefix');
+    if (!this.name) missingProps.push('name');
+    if (!this.familyStylePathSegment) missingProps.push('familyStylePathSegment');
+    if (!this.svgFetchBaseUrl) missingProps.push('svgFetchBaseUrl');
+    if (!this.kitToken) missingProps.push('kitToken');
+    if (!this.getUrlText) missingProps.push('getUrlText');
+
+    if (missingProps.length > 0) {
+      console.error(`${CONSOLE_MESSAGE_PREFIX}: Missing required props:`, missingProps);
       return;
     }
 
-    if (!this.familyStylePathSegment) {
-      console.error(`${CONSOLE_MESSAGE_PREFIX}: fa-icon: the 'familyStylePathSegment' prop is required to render this icon but not provided.`, this);
+    // Validate SVG API availability
+    const { findIconDefinition, library } = this.svgApi || {};
+    if (!findIconDefinition || !library?.add) {
+      console.error(`${CONSOLE_MESSAGE_PREFIX}: Invalid svgApi - missing findIconDefinition or library.add`);
       return;
     }
 
-    const { findIconDefinition } = this.svgApi;
-
-    const iconDefinition =
-      findIconDefinition &&
-      findIconDefinition({
-        prefix: this.stylePrefix,
-        iconName: this.name,
-      });
+    // Try to find icon definition in existing library
+    const iconDefinition = findIconDefinition({
+      prefix: this.stylePrefix,
+      iconName: this.name,
+    });
 
     if (iconDefinition) {
       this.setIconDefinition(iconDefinition);
       return;
     }
 
-    if (!this.pro) {
-      console.error(`${CONSOLE_MESSAGE_PREFIX}: fa-icon: 'pro' prop is false but no free icon is available`, this);
-      return;
-    }
+    // Construct the icon URL and set up processing function
+    const iconUrl = `${this.svgFetchBaseUrl}/${this.familyStylePathSegment}/${this.name}${this.svgFetchBaseUrl.includes('svg-objects') ? '.json' : '.svg'}?token=${this.kitToken}`;
 
-    if (!this.svgFetchBaseUrl) {
-      console.error(`${CONSOLE_MESSAGE_PREFIX}: fa-icon: 'svgFetchBaseUrl' prop is absent but is necessary for fetching icon`, this);
-      return;
-    }
+    const processIconDefinition = (response: string) => {
+      if (!response) {
+        throw new Error('Empty response received from fetch');
+      }
+      let iconDefinition;
 
-    if (!this.kitToken) {
-      console.error(`${CONSOLE_MESSAGE_PREFIX}: fa-icon: 'kitToken' prop is absent but is necessary for accessing icon`, this);
-      return;
-    }
+      try {
+        const parsed = this.svgFetchBaseUrl.includes('svg-objects') ? JSON.parse(response) : parseSvgText(response);
 
-    this.loading = true;
+        if (!parsed) {
+          throw new Error('Failed to parse response');
+        }
 
-    const iconUrl = `${this.svgFetchBaseUrl}/${this.familyStylePathSegment}/${this.name}.svg?token=${this.kitToken}`;
+        const iconData = Array.isArray(parsed) ? parsed : parsed.icon;
```

Comment:
Edited... now `parsed` is an array (when coming from `parseSvgText()` 
but an object when coming from `JSON.parse` with a JSON file.